xscen/extract.py

```diff
@@ -387,6 +406,23 @@ def extract_dataset(
 
         out_dict[xrfreq] = ds
 
+    if mask:
+        if isinstance(mask, xr.Dataset):
+            ds_mask = mask["mask"]
+        elif (
+            "fx" in out_dict and "mask" in out_dict["fx"]
+        ):  # get mask that was extracted above
+            ds_mask = out_dict["fx"]["mask"].copy()
+        else:
+            raise ValueError(
+                "No mask found. Either pass a xr.Dataset to the `mask` argument or pass a `dc` that includes a dataset with variable `mask`."
+            )
+
+        # iter over all xrfreq to apply the mask
+        for xrfreq, ds in out_dict.items():
+            if xrfreq != "fx":
```

Comment:
I think we want to subset fixed fields too?

Reply:
Well we for sure don't want to mask the mask and I assumed we didn't want to mask other fixed fields but if you think there is a need for it I can make it avoid just the mask.

Reply:
I'm not 100% sure but if I ask for a mask, I'd expect it to be applied to everything?

Reply:
yeah you are right